IdentityCore/tests/automation/MSIDTestAccountsProvider.m

```diff
@@ -0,0 +1,170 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDTestAccountsProvider.h"
+#import "MSIDAutomation-Swift.h"
+
+@interface MSIDTestAccountsProvider()
+
+@property (nonatomic, strong) NSMutableDictionary *cachedConfigurations;
+@property (nonatomic, strong) KeyvaultAuthentication *keyvaultAuthentication;
+@property (nonatomic, strong) NSString *apiPath;
+
+@end
+
+@implementation MSIDTestAccountsProvider
+
+- (instancetype)initWithClientCertificateContents:(NSString *)certificate
+                              certificatePassword:(NSString *)password
+                         additionalConfigurations:(NSDictionary *)additionalConfigurations
+                                          apiPath:(NSString *)apiPath
+{
+    self = [super init];
+
+    if (self)
+    {
+        _cachedConfigurations = [NSMutableDictionary dictionary];
+        _keyvaultAuthentication = [[KeyvaultAuthentication alloc] initWithCertContents:certificate certPassword:password];
+        _apiPath = apiPath;
+        [_cachedConfigurations addEntriesFromDictionary:additionalConfigurations];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithConfigurationPath:(NSString *)configurationPath
+{
+    NSData *configurationData = [NSData dataWithContentsOfFile:configurationPath];
+
+    if (!configurationData)
+    {
+        return nil;
+    }
+
+    NSError *jsonError = nil;
+    NSDictionary *configurationDictionary = [NSJSONSerialization JSONObjectWithData:configurationData options:0 error:&jsonError];
+
+    if (jsonError || !configurationDictionary)
+    {
+        return nil;
+    }
+
+    NSString *apiPath = configurationDictionary[@"api_path"];
+    NSString *certificatePassword = configurationDictionary[@"certificate_password"];
+    NSString *encodedCertificate = configurationDictionary[@"certificate_data"];
+
+    NSArray *additionalConfs = configurationDictionary[@"additional_confs"];
+
+    NSMutableDictionary *additionalConfsDictionary = [NSMutableDictionary dictionary];
+
+    for (NSDictionary *additionalConf in additionalConfs)
+    {
+        NSDictionary *requestDict = additionalConf[@"request"];
+        NSDictionary *configurationDict = additionalConf[@"configuration"];
+
+        MSIDTestConfigurationRequest *request = [MSIDTestConfigurationRequest requestWithDictionary:requestDict];
+        MSIDTestConfiguration *configuration = [[MSIDTestConfiguration alloc] initWithJSONDictionary:configurationDict];
+        additionalConfsDictionary[request] = configuration;
```

Comment:
Q: do we need a nil check here on the configuration value?

Reply:
If conf is nil for some reason, it's not of much use for us anyway, so I don't see a reason for a nil check